src/main/java/org/opensearch/timeseries/model/Config.java

```diff
@@ -357,6 +360,7 @@ protected Config(
         this.flattenResultIndexMapping = Strings.trimToNull(resultIndex) == null ? null : flattenResultIndexMapping;
         this.lastUIBreakingChangeTime = lastBreakingUIChangeTime;
         this.frequency = frequency;
+        this.autoCreated = autoCreated != null ? autoCreated : false;
```

Comment:
During b/g, say a new node creates a new detector whose autoCreated null and then changed to default false. In a old node, it does not handle the autoCreated field and collapse. Should we only emit the field when true to keep bwc?